lesson16/main.js

```diff
@@ -65,133 +26,164 @@ async function fetchData(url) {
+function createArticles(data) {
+  const articleList = data.article;
+  console.log(articleList);
+  for (const article of articleList) {
+    const articleContainer = document.createElement("div");
+    articleContainer.classList.add("article-container");
+
+    const title = document.createElement("li");
+    const titleAnchor = document.createElement("a");
+    titleAnchor.href = "#";
+    titleAnchor.textContent = article.title;
+    title.appendChild(titleAnchor);
+
+    const newIconContainer = document.createElement("div");
+
+    const articleDate = new Date(article.date);
+    if (withinThreeDays(articleDate)) {
+      const newIcon = document.createElement("img");
+      newIcon.src = "./img/new.png";
+      newIcon.classList.add("new");
+      newIconContainer.appendChild(newIcon);
+    }
+
+    articleContainer.appendChild(title);
+    articleContainer.appendChild(newIconContainer);
+
+    fragmentTitles.appendChild(articleContainer);
+  }
 }
 
-function resetValidation() {
-  isValidateName = false;
-  isValidateNumber = false;
+function createImage(data) {
+  const img = document.createElement("img");
+  img.classList.add("article-image");
+  img.src = data.image;
+  img.width = "100";
+  img.height = "100";
+  fragmentImages.appendChild(img);
+  return img;
 }
 
-function checkEnableSubmit() {
-  if (isValidateName && isValidateNumber) {
-    validInput();
-  } else {
-    invalidInput();
+function combineArticleImage(data, img) {
+  const genreContainer = document.createElement("div");
+  genreContainer.classList.add("genre-container");
+  const titleArea = document.createElement("div");
+  const imageArea = document.createElement("div");
+  titleArea.classList.add("content", "title-area");
+  imageArea.classList.add("content", "image-area");
+  genreContainer.id = data.id;
+  genreContainer.appendChild(titleArea).appendChild(fragmentTitles);
+  genreContainer.appendChild(imageArea).appendChild(img);
+
+  //初期表示の設定
+  if (data.select) {
+    genreContainer.classList.add("active");
   }
+  fragmentGenres.appendChild(genreContainer);
 }
 
+function removeCircle() {
+  document.getElementById("loading-circle").remove();
+}
 
-function invalidInput(errorMessage) {
-  promptMessage.textContent = errorMessage;
-  promptMessage.style.color = "red";
-  fetchButton.disabled = true;
+function withinThreeDays(day) {
+  const today = new Date();
+  const msInThreeDays = 20 * 24 * 60 * 60 * 1000;
+  const diff = today.getTime() - day.getTime();
+  return diff < msInThreeDays;
 }
 
-function validInput() {
-  promptMessage.textContent = "入力後、取得ボタンを押してね";
-  promptMessage.style.color = "black";
-  fetchButton.disabled = false;
+function addClickEventListener(tabs, contents) {
+  tabArea.addEventListener("click", (e) => {
+    const targetElement = e.target;
+    if (targetElement.tagName.toLowerCase() === "a") {
+      tabs.forEach((tab) => {
+        tab.classList.remove("active");
+        console.log(tab);
+      });

```

Comment:
IMO
You can also write like this
```suggestion
      tabArea.querySelector(".active").classList.remove("active");
```

Reply:
Thank you sae-san :)
It's very simple!
ffebfec220d44538c2b5e189a2ac2c530bdf010a